diff --git a/src/misc1.c b/src/misc1.c
index 2a2d9cf64..31304b9ed 100644
--- a/src/misc1.c
+++ b/src/misc1.c
@@ -9230,15 +9230,21 @@ lisp_match(char_u *p)
     char_u	buf[LSIZE];
     int		len;
     char_u	*word = *curbuf->b_p_lw != NUL ? curbuf->b_p_lw : p_lispwords;
+    int flipsense = 0;
+
+    if (*word == '!') {
+        flipsense = 1;
+        word += 1;
+    }
 
     while (*word != NUL)
     {
 	(void)copy_option_part(&word, buf, LSIZE, ",");
 	len = (int)STRLEN(buf);
 	if (STRNCMP(buf, p, len) == 0 && p[len] == ' ')
-	    return TRUE;
+	    return !flipsense;
     }
-    return FALSE;
+    return !!flipsense;
 }
 
 /*
@@ -9262,7 +9268,7 @@ get_lisp_indent(void)
 {
     pos_T	*pos, realpos, paren;
     int		amount;
-    char_u	*that;
+    char_u	*firstthat, *that;
     colnr_T	col;
     colnr_T	firsttry;
     int		parencount, quotecount;
@@ -9294,9 +9300,9 @@ get_lisp_indent(void)
 	{
 	    if (linewhite(curwin->w_cursor.lnum))
 		continue;
-	    for (that = ml_get_curline(); *that != NUL; ++that)
+	    for (firstthat = that = ml_get_curline(); *that != NUL; ++that)
 	    {
-		if (*that == ';')
+		if (*that == ';' && (that == firstthat || *(that-1) != '#'))
 		{
 		    while (*(that + 1) != NUL)
 			++that;
diff --git a/src/normal.c b/src/normal.c
index ce8ca531a..2bcef6a1d 100644
--- a/src/normal.c
+++ b/src/normal.c
@@ -1180,7 +1180,7 @@ getcount:
     /*
      * If an operation is pending, handle it...
      */
-    do_pending_operator(&ca, old_col, FALSE);
+    do_pending_operator(&ca, old_col, FALSE, old_pos);
 
     /*
      * Wait for a moment when a message is displayed that will be overwritten
@@ -1359,7 +1359,7 @@ set_vcount_ca(cmdarg_T *cap, int *set_prevcount)
  * Handle an operator after visual mode or when the movement is finished
  */
     void
-do_pending_operator(cmdarg_T *cap, int old_col, int gui_yank)
+do_pending_operator(cmdarg_T *cap, int old_col, int gui_yank, pos_T old_pos)
 {
     oparg_T	*oap = cap->oap;
     pos_T	old_cursor;
@@ -1889,6 +1889,7 @@ do_pending_operator(cmdarg_T *cap, int old_col, int gui_yank)
 		curwin->w_p_lbr = lbr_saved;
 #endif
 		(void)op_yank(oap, FALSE, !gui_yank);
+		if(p_ylc) curwin->w_cursor = old_pos;
 	    }
 	    check_cursor_col();
 	    break;
@@ -6720,7 +6721,7 @@ nv_brackets(cmdarg_T *cap)
 		    cap->nchar = NUL;
 		    cap->oap->regname = regname;
 		    nv_operator(cap);
-		    do_pending_operator(cap, 0, FALSE);
+		    do_pending_operator(cap, 0, FALSE, old_pos);
 		}
 		if (VIsual_active)
 		{
@@ -9066,7 +9067,7 @@ nv_edit(cmdarg_T *cap)
 	cap->cmdchar = 'd';
 	cap->nchar = NUL;
 	nv_operator(cap);
-	do_pending_operator(cap, 0, FALSE);
+	do_pending_operator(cap, 0, FALSE, old_pos);
 	cap->cmdchar = K_PS;
 
 	/* When the last char in the line was deleted then append. Detect this
@@ -9382,6 +9383,7 @@ nv_put(cmdarg_T *cap)
     int		was_visual = FALSE;
     int		dir;
     int		flags = 0;
+    pos_T       old_pos = curwin->w_cursor;		/* remember where cursor was */
 
     if (cap->oap->op_type != OP_NOP)
     {
@@ -9435,7 +9437,7 @@ nv_put(cmdarg_T *cap)
 	    cap->nchar = NUL;
 	    cap->oap->regname = NUL;
 	    nv_operator(cap);
-	    do_pending_operator(cap, 0, FALSE);
+	    do_pending_operator(cap, 0, FALSE, old_pos);
 	    empty = (curbuf->b_ml.ml_flags & ML_EMPTY);
 
 	    /* delete PUT_LINE_BACKWARD; */
diff --git a/src/ops.c b/src/ops.c
index d23b3077f..76cb61758 100644
--- a/src/ops.c
+++ b/src/ops.c
@@ -6453,7 +6453,7 @@ clip_get_selection(VimClipboard *cbd)
 	ca.cmdchar = 'y';
 	ca.count1 = 1;
 	ca.retval = CA_NO_ADJ_OP_END;
-	do_pending_operator(&ca, 0, TRUE);
+	do_pending_operator(&ca, 0, TRUE, old_op_start);
 	y_previous = old_y_previous;
 	y_current = old_y_current;
 	curwin->w_cursor = old_cursor;
diff --git a/src/option.c b/src/option.c
index 32e97c99b..f8115592e 100644
--- a/src/option.c
+++ b/src/option.c
@@ -2990,6 +2990,10 @@ static struct vimoption options[] =
 			    (char_u *)&p_wd, PV_NONE,
 			    {(char_u *)0L, (char_u *)0L} SCRIPTID_INIT},
 
+    {"yankleavecursor",  NULL,   P_NUM|P_VI_DEF,
+			    (char_u *)&p_ylc, PV_NONE,
+			    {(char_u *)FALSE, (char_u *)0L} SCRIPTID_INIT},
+
 /* terminal output codes */
 #define p_term(sss, vvv)   {sss, NULL, P_STRING|P_VI_DEF|P_RALL|P_SECURE, \
 			    (char_u *)&vvv, PV_NONE, \
diff --git a/src/option.h b/src/option.h
index 62e66cd3d..cbe0a4cae 100644
--- a/src/option.h
+++ b/src/option.h
@@ -974,6 +974,7 @@ EXTERN int	p_write;	/* 'write' */
 EXTERN int	p_wa;		/* 'writeany' */
 EXTERN int	p_wb;		/* 'writebackup' */
 EXTERN long	p_wd;		/* 'writedelay' */
+EXTERN long	p_ylc;		/* 'yankleavecursor' */
 
 /*
  * "indir" values for buffer-local opions.
diff --git a/src/proto/normal.pro b/src/proto/normal.pro
index 55d12bb29..49d802394 100644
--- a/src/proto/normal.pro
+++ b/src/proto/normal.pro
@@ -1,7 +1,7 @@
 /* normal.c */
 void init_normal_cmds(void);
 void normal_cmd(oparg_T *oap, int toplevel);
-void do_pending_operator(cmdarg_T *cap, int old_col, int gui_yank);
+void do_pending_operator(cmdarg_T *cap, int old_col, int gui_yank, pos_T old_pos);
 int do_mouse(oparg_T *oap, int c, int dir, long count, int fixindent);
 void check_visual_highlight(void);
 void end_visual_mode(void);
diff --git a/src/search.c b/src/search.c
index 36410e50f..2e7952591 100644
--- a/src/search.c
+++ b/src/search.c
@@ -2520,7 +2520,8 @@ check_linecomment(char_u *line)
 				      && *(p - 1) != '\\' && *(p - 2) != '#'))
 			in_str = TRUE;
 		}
-		else if (!in_str && ((p - line) < 2
+		else if (!in_str && ((p - line) < 1 || *(p - 1) != '#') &&
+			    ((p - line) < 2
 				    || (*(p - 1) != '\\' && *(p - 2) != '#')))
 		    break;	/* found! */
 		++p;
