*** normal.pro.0	2010-08-15 08:30:45.000000000 -0400
--- normal.pro	2010-11-27 22:11:04.000000000 -0500
***************
*** 1,7 ****
  /* normal.c */
  void init_normal_cmds __ARGS((void));
  void normal_cmd __ARGS((oparg_T *oap, int toplevel));
! void do_pending_operator __ARGS((cmdarg_T *cap, int old_col, int gui_yank));
  int do_mouse __ARGS((oparg_T *oap, int c, int dir, long count, int fixindent));
  void check_visual_highlight __ARGS((void));
  void end_visual_mode __ARGS((void));
--- 1,7 ----
  /* normal.c */
  void init_normal_cmds __ARGS((void));
  void normal_cmd __ARGS((oparg_T *oap, int toplevel));
! void do_pending_operator __ARGS((cmdarg_T *cap, int old_col, int gui_yank, pos_T old_pos));
  int do_mouse __ARGS((oparg_T *oap, int c, int dir, long count, int fixindent));
  void check_visual_highlight __ARGS((void));
  void end_visual_mode __ARGS((void));
