*** misc1.c.0	2010-08-15 07:24:05.000000000 -0400
--- misc1.c	2010-12-17 14:54:59.718965381 -0500
***************
*** 8194,8208 ****
      char_u	buf[LSIZE];
      int		len;
      char_u	*word = p_lispwords;
  
      while (*word != NUL)
      {
  	(void)copy_option_part(&word, buf, LSIZE, ",");
  	len = (int)STRLEN(buf);
  	if (STRNCMP(buf, p, len) == 0 && p[len] == ' ')
! 	    return TRUE;
      }
!     return FALSE;
  }
  
  /*
--- 8194,8214 ----
      char_u	buf[LSIZE];
      int		len;
      char_u	*word = p_lispwords;
+     int flipsense = 0;
+ 
+     if (*word == '!') {
+         flipsense = 1;
+         word += 1;
+     }
  
      while (*word != NUL)
      {
  	(void)copy_option_part(&word, buf, LSIZE, ",");
  	len = (int)STRLEN(buf);
  	if (STRNCMP(buf, p, len) == 0 && p[len] == ' ')
! 	    return flipsense ? FALSE : TRUE;
      }
!     return flipsense ? TRUE : FALSE;
  }
  
  /*
***************
*** 8226,8232 ****
  {
      pos_T	*pos, realpos, paren;
      int		amount;
!     char_u	*that;
      colnr_T	col;
      colnr_T	firsttry;
      int		parencount, quotecount;
--- 8232,8238 ----
  {
      pos_T	*pos, realpos, paren;
      int		amount;
!     char_u	*firstthat, *that;
      colnr_T	col;
      colnr_T	firsttry;
      int		parencount, quotecount;
***************
*** 8258,8266 ****
  	{
  	    if (linewhite(curwin->w_cursor.lnum))
  		continue;
! 	    for (that = ml_get_curline(); *that != NUL; ++that)
  	    {
! 		if (*that == ';')
  		{
  		    while (*(that + 1) != NUL)
  			++that;
--- 8264,8272 ----
  	{
  	    if (linewhite(curwin->w_cursor.lnum))
  		continue;
! 	    for (firstthat = that = ml_get_curline(); *that != NUL; ++that)
  	    {
! 		if (*that == ';' && (that == firstthat || *(that-1) != '#'))
  		{
  		    while (*(that + 1) != NUL)
  			++that;
